db-scheduler/src/main/java/com/github/kagkarlsson/scheduler/SchedulerClient.java

```diff
@@ -352,6 +380,26 @@ public <T> void reschedule(TaskInstanceId taskInstanceId, Instant newExecutionTi
       }
     }
 
+    @Override
+    public <T> void rescheduleOrCreate(TaskInstance<T> taskInstance, Instant executionTime) {
+      Optional<ScheduledExecution<Object>> executionObject =
+          this.getScheduledExecutions().stream()
+              .filter(e -> e.getTaskInstance().getId().equals(taskInstance.getId()))
+              .findFirst();
+      if (executionObject.isPresent()) {
+        this.reschedule(taskInstance, executionTime, taskInstance.getData());
+      } else {
+        this.scheduleIfNotExists(taskInstance, executionTime);
+      }
+    }
+
+    @Override
+    public <T> void rescheduleOrCreate(SchedulableInstance<T> schedulableInstance) {
+      rescheduleOrCreate(
+          schedulableInstance.getTaskInstance(),
+          schedulableInstance.getNextExecutionTime(clock.now()));
+    }
```

Comment:
Agreed that it is hard to come up with a good name for this operation.. 🤔 

rescheduleOrCreate
scheduleOrReschedule
scheduleOrMove
scheduleOrReplace
schedule(id, WhenExists.RESCHEDULE).  (OVERWRITE/DO_NOTHING...)
..

What do you think about a variant of the last one, where you have a parameter for what to do when the task-instance already exists?

Reply:
schedule as the method name with an additional parameter seems most intuitive to me. 

I have therefore adapted the code accordingly. 
schedule now also returns a boolean value, just like scheduleIfNotExists and reschedule.